File: 2537. Count the Number of Good Subarrays/Solution.py
class Solution:
    def countGood(self, nums: List[int], k: int) -> int:
        res, pairs, right, n = 0, 0, 0, len(nums)
        counter = dict()
        for left in range(n):
            while right < n and pairs < k:
                if nums[right] not in counter: counter[nums[right]] = 0
                pairs += counter[nums[right]]
                counter[nums[right]] += 1
                right += 1
            if pairs >= k:
                res += n - right + 1
            pairs -= counter[nums[left]] - 1
            counter[nums[left]] -= 1
        return res

File: 2537. Count the Number of Good Subarrays/Solution.java
class Solution {
    public long countGood(int[] nums, int k) {
        long res = 0, pairs = 0;
        int n = nums.length;
        Map<Integer, Integer> counter = new HashMap<>();
        for (int left = 0, right = 0; left < n; left++) {
            while (right < n && pairs < k) {
                pairs += counter.getOrDefault(nums[right], 0);
                counter.put(nums[right], counter.getOrDefault(nums[right], 0) + 1);
                right++;
            }
            if (pairs >= k) {
                res += n - right + 1;
            }
            pairs -= counter.get(nums[left]) - 1;
            counter.put(nums[left], counter.get(nums[left]) - 1);
        }
        return res;
    }
}